### PR TITLE
added update-packages feature

### PR DESCRIPTION
**Pull request by esunar**
fixed missing docs
consider case where model_mapping is empty or None added unit tests for update-packages
added extra unit tests for corner cases

I am closing #44 since it is polluted.

### CODE REVIEW

File: docs/commands/update-packages/update-packages.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+```json
+[
+ {
+  "context": { ... },
+  "success": true,
+  "output": {
+   "model1": {
+    "applications": [
+     { # [1],
```
Comment (rgildein) [inline docs/commands/update-packages/update-packages.md:96]: not-blocking: can you update this format, so GitHub can previewed it properly

File: juju_spell/cli/update_packages.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            Application(
+                name_expr=app["application"],
+                dist_upgrade=dist_upgrade,
+                results=[],
+                packages_to_update=packages_to_update,
+            )
+        )
+
+    if len(errors) > 0:
+        raise JujuSpellError("errors in input file:\n" + "\n".join(errors))
```
Comment (rgildein) [inline juju_spell/cli/update_packages.py:127]: not-blocking: we should use `os.linesep`

**Review by rgildein — APPROVED** [review]
two not-blocking comments

File: juju_spell/cli/update_packages.py (hunk 2 above)
Comment (esunar) [reply juju_spell/cli/update_packages.py]: done

File: docs/commands/update-packages/update-packages.md (hunk 1 above)
Comment (esunar) [reply docs/commands/update-packages/update-packages.md]: json does not support comments, this the best I can do

Comment (esunar) [reply docs/commands/update-packages/update-packages.md]: I found a better way if you format code as javascript you can add comments

**Review by rgildein — APPROVED** [review]
LGTM
